### DISCUSSION TITLE
gatsby-plugin-manifest: Include required config in gatsby-config.js

### DISCUSSION THREAD

**Discussion by harry643** [gatsbyjs/gatsby#28504] [Ideas / Feature Requests] [answered]
I'm a Gatsby beginner. I was going through the Gatsby interactive command-line setup and chose the 'Generate a manifest' option. After setup, I run `gatsby develop` and I get an error:

![image](https://user-images.githubusercontent.com/8885267/101295790-af37b900-3817-11eb-91f3-1f1e1ba9e91e.png)

This is because the required config that gatsby-plugin-manifest needs is not in gatsby-config.js. It is currently this that causes the problem:

```
module.exports = {
 plugins: ["gatsby-plugin-manifest"],
};
```

My proposal is to include a basic config in gatsby-config.js if you choose the 'Generate a manifest' option during setup so you can run the project with no problems. Like this from the [gatsby-plugin-manifest page](https://www.gatsbyjs.com/plugins/gatsby-plugin-manifest/?=#add-plugin-and-manifest-settings---required):

```
module.exports = {
plugins: [
{
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `GatsbyJS`,
        short_name: `GatsbyJS`,
        start_url: `/`,
        background_color: `#f7f0eb`,
        theme_color: `#a2466c`,
        display: `standalone`,
        icon: `./src/images/icon.png`,
      }
```

This does require creation of the images folder and a default icon image.

**Comment by LekoArts** [comment, marked as answer]
Thanks for highlighting that, we're aware of it and track it as a bug now here: https://github.com/gatsbyjs/gatsby/issues/28613
